src/adapters/osm_schedule.js

```diff
@@ -0,0 +1,124 @@
+const strftime = require('strftime')
+const days = ['mo', 'tu', 'we', 'th', 'fr', 'sa', 'su'];
+
+
+function OsmSchedule(scheduleResponse, messages) {
+  this.schedule = parse(scheduleResponse.raw)
+  this.displayHours = translateSchedule(this.schedule)
+  this.nextTransition = nextTransitionTime(scheduleResponse.seconds_before_next_transition, scheduleResponse.next_transition_datetime)
+  this.status = scheduleStatus(this.schedule, messages)
+}
+
+/* private */
+
+function parse(rawSchedule) {
+  var result = null;
+  if (rawSchedule === '' || rawSchedule === null) {
+    result = null;
+  }
+  else if (rawSchedule === '24/7') {
+    result = {isTwentyForSeven: true};
+  }
+  else if (rawSchedule === 'seasonal') {
+    result = {'seasonal': true};
+  }
+  else {
+    result = {};
+    var modified_some_days = false;
+    for (var k = 0; k < days.length; k++) {
+      result[days[k]] = null;
+    }
+
+    var dayregex = /^(mo|tu|we|th|fr|sa|su)\-?(mo|tu|we|th|fr|sa|su)?$/,
+      timeregex = /^\s*(\d\d:\d\d)\-(\d\d:\d\d)\s*$/,
+      dayranges = rawSchedule.toLowerCase().split(/\s*;\s*/),
+      dayrange;
+    while((dayrange = dayranges.shift())) {
+      var daytimes = dayrange.trim().split(/\s+/),
+        daytime,
+        startday = 0,
+        endday = 6,
+        whichDays,
+        whichTimes,
+        starttime,
+        endtime;
+
+      while((daytime = daytimes.shift())) {
+        if (dayregex.test(daytime)) {
+          var daymatches = daytime.match(dayregex);
+
+          if (daymatches.length === 3) {
+            startday = days.indexOf(daymatches[1]);
+            if (daymatches[2]) {
+              endday = days.indexOf(daymatches[2]);
+            } else {
+              endday = startday;
+            }
+          } else {
+            return null;
+          }
+        } else if (timeregex.test(daytime)) {
+          var timematches = daytime.match(timeregex);
+
+          if (timematches.length === 3) {
+            starttime = timematches[1];
+            endtime = timematches[2];
+          } else {
+            return null;
+          }
+        } else {
+          return null;
+        }
+      }
+
+      for (var j = startday; j <= endday; j++) {
+        result[days[j]] = [starttime, endtime];
+        modified_some_days = true;
+      }
+
+      if (!modified_some_days) {
+        result = null;
+      }
+    }
+  }
+
+  return result;
+}
+
+
+function scheduleStatus(schedule, timeMessages) {
+  if(!schedule) {
+    return {msg : '', color : '#fff'}
+  }
+  let remaining = schedule.seconds_before_next_transition
```

Comment:
I think there is a confusion here between `scheduleResponse` and `schedule` that has no "seconds_before_next_transition" defined. And as a result, the status is now always "open" :stuck_out_tongue: 

Also, maybe we could completely get rid of the client-side `parse` function, now that the API returns complete "days" schedules (with potentially multiple opening periods per day).